### PR TITLE
Update Episerver packages, target IndexingService to .NET 7

### PR DESCRIPTION
**Pull request by karolberezicki**
Hi 🙂 

In this PR I would like to bring following changes to `content-search-lucene`:
* Bring `Episerver` dependencies up to latest versions (`12.13.1`/`12.17.1`)
* Target `IndexingService` to `.NET 7`
* Set `ContentCacheScope` for `IndexingJob` to reduce memory consumption for larger project
* Set SDK to 7.0
* Build Docker image with .NET 7 by default
* Minor updates to test packages

I plan to bring more memory optimizations also for data collecting in `IndexingJob` / `ContentSearchHandlerImplementation` , but in a separate PR, if this will be approved and merged.

### CODE REVIEW

**Comment by lunchin** [comment]
The cla is now enabled on this repo, please sign the cla for us to be able to except your pull requests

File: Directory.Build.props [hunk 1]
```diff
@@ -2,9 +2,8 @@
   <Import Project="build\version.props" />
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
-    <LangVersion>9.0</LangVersion>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
```
Comment (lunchin) [inline Directory.Build.props:5]: We are not specifically targeting standard releases moving forward just LTS.  We test to make sure package can run on latest but do update target frameworks

File: src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj [hunk 2]
```diff
@@ -1,16 +1,15 @@
 <Project Sdk="Microsoft.NET.Sdk.Razor">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
```
Comment (lunchin) [inline src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj:3]: Keep at 6

File: src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
   <ItemGroup>
     <None Remove="Resources\language.xml" />
   </ItemGroup>
   <ItemGroup>
     <EmbeddedResource Include="Resources\language.xml" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="EPiServer.CMS.UI.Core" />
-    <PackageReference Include="EPiServer.CMS.AspNetCore.Templating" />
+    <PackageReference Include="EPiServer.CMS.UI.Core" Version="12.9.3" />
```
Comment (lunchin) [inline src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj:12]: Should use dependency versions from props file

File: src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
+    <PackageReference Include="EPiServer.CMS.AspNetCore.Templating" Version="12.13.2"  />
```
Comment (lunchin) [inline src/EPiServer.Search.Cms/EPiServer.Search.Cms.csproj:13]: Should use dependency versions from props file

File: src/EPiServer.Search.IndexingService/EPiServer.Search.IndexingService.csproj [hunk 5]
```diff
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
```
Comment (lunchin) [inline src/EPiServer.Search.IndexingService/EPiServer.Search.IndexingService.csproj:3]: Just target 7 for the container app

**Review by lunchin — CHANGES_REQUESTED** [review]

**Comment by karolberezicki** [comment]
Thank you for the review @lunchin, much appreciated :) 
I've retargeted the packages and framework versions according to the comments.

**Review by lunchin — APPROVED** [review]

**Comment by lunchin** [comment]
create new pr to fix build issues on fork

**Pull request closed without merging**